metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py

```diff
@@ -93,35 +92,35 @@ def get_workunits(
         if self.connection is None:
             return
 
-        self._populate_external_lineage_map()
+        self._populate_external_lineage_map(discovered_tables)
 
         if self.config.include_view_lineage:
             if len(discovered_views) > 0:
                 yield from self.get_view_upstream_workunits(discovered_views)
-                logger.info(
-                    f"Upstream lineage detected for {self.report.num_views_with_upstreams} views.",
-                )
             else:
                 logger.info("No views found. Skipping View Lineage Extraction.")
 
         yield from self.get_table_upstream_workunits(discovered_tables)
-        logger.info(
-            f"Upstream lineage detected for {self.report.num_tables_with_upstreams} tables.",
-        )
 
-        yield from self.get_table_external_upstream_workunits()
+        if self._external_lineage_map:  # Some external lineage is yet to be emitted
+            yield from self.get_table_external_upstream_workunits(discovered_tables)
 
-    def get_table_external_upstream_workunits(self) -> Iterable[MetadataWorkUnit]:
-        if self._external_lineage_map:
-            for (
-                dataset_name,
-                external_lineage,
-            ) in self._external_lineage_map.items():
-                upstreams = self.get_external_upstreams(external_lineage)
-                if upstreams:
-                    yield self._create_upstream_lineage_workunit(
-                        dataset_name, upstreams
-                    )
+    def get_table_external_upstream_workunits(
+        self, discovered_tables: List[str]
+    ) -> Iterable[MetadataWorkUnit]:
+        for (
+            dataset_name,
+            external_lineage,
+        ) in self._external_lineage_map.items():
+            if dataset_name not in discovered_tables:
+                continue
```

Comment:
This is repetitive right? `_populate_external_lineage_map` does the filtering already

Reply:
removed this.